### PR TITLE
add zn_screen_layout

### PR DESCRIPTION
**Pull request by watasuke102**
add screen layout, and screens are now owned by zn_screen_layout instead of zn_scene.

zn_scene -> zn_screen_layout -> zn_screen (wl_list)

### CODE REVIEW

**Comment by watasuke102** [comment]
maybe implemented? Is this ok?

**Comment by Aki-7** [comment]
ref: https://github.com/zigen-project/zen/issues/86

File: zen/scene/screen-layout.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  struct zn_screen* screen;
+  struct wlr_box box;
+  wl_list_for_each(screen, &self->screens, link)
+  {
+    wlr_output_effective_resolution(
+        screen->output->wlr_output, &box.width, &box.height);
+    screen->x = x;
+    screen->y = 0;
+    x += box.width;
+  }
```
Comment (Aki-7) [inline zen/scene/screen-layout.c:21]: int x doesn't seem necessary.
I want you to declare variable at the beginning of the scope.
```suggestion
  struct zn_screen* screen;
  struct wlr_box box;
  wl_list_insert(&self->screens, &new_screen->link);
  wl_list_for_each(screen, &self->screens, link)
  {
    wlr_output_effective_resolution(
        screen->output->wlr_output, &box.width, &box.height);
    screen->x += box.width;
    screen->y = 0;
  }
```

File: include/zen/scene/screen-layout.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+struct zn_screen;
+
+struct zn_screen_layout {
+  struct wl_list screens;  // zn_screen::link
+};
+
+void zn_screen_layout_add(
+    struct zn_screen_layout* self, struct zn_screen* screen);
+void zn_screen_layout_remove(
+    struct zn_screen_layout* self, struct zn_screen* screen);
```
Comment (Aki-7) [inline include/zen/scene/screen-layout.h:16]:
```suggestion
void zn_screen_layout_add(
    struct zn_screen_layout* self, struct zn_screen* screen);

void zn_screen_layout_remove(
    struct zn_screen_layout* self, struct zn_screen* screen);
```

**Review by Aki-7 — APPROVED** [review]

**Pull request merged**